packages/verkle/src/verkleTree.ts

```diff
@@ -255,6 +255,16 @@ export class VerkleTree {
       this.DEBUG && this.debug(`Creating new leaf node at stem: ${bytesToHex(stem)}`, ['PUT'])
     }
     // Update value in leaf node and push to putStack
+    if (equalsBytes(value, createDeletedLeafValue())) {
+      // Special case for when the deleted leaf value is passed to `put`
+      // Writing the deleted leaf value to the suffix indicated in the key
+      this.DEBUG &&
+        this.debug(
+          `Deleting value at suffix: ${suffix} in leaf node with stem: ${bytesToHex(stem)}`,
+          ['DEL']
+        )
+      leafNode.setValue(suffix, VerkleLeafNodeValue.Deleted)
+    }
     leafNode.setValue(suffix, value)
```

Comment:
Isn't line 268 overwriting what has been done in line 266 within the if block?

Reply:
Indeed.  Will address in the next commit.